datafusion/expr/src/expr_schema.rs

```diff
@@ -123,7 +123,8 @@ impl ExprSchemable for Expr {
                         Ok(field.data_type().clone())
                     }
                     DataType::Struct(_) => {
-                        not_impl_err!("unnest() does not support struct yet")
+                        // TODO: this is not correct, because unnest(struct) wll result into multiple data_type
```

Comment:
I'm still finding a way to solve this TODO

Reply:
When do we end up with an `unest` **Expr**  (vs an `LogicalPlan::Unnest`). 

Could we simply return `not_yet_implemented` error here?

Reply:
if we returns error here, the error will stop the query plan from continue.
Initially unnest comes into projection step as an Expr here: https://github.com/apache/datafusion/blob/d58bae487329b7a7078429f083bffc611f42c8c7/datafusion/expr/src/utils.rs#L734, where it check data types of each expr, then later on we call
[try_process_unnest](https://github.com/apache/datafusion/blob/ae4b3a0f8366ab18be5ef5cfa2b3cc3aca12baf1/datafusion/sql/src/select.rs#L226) to do the transformation from unnest(struct) -> struct.field1, struct.field 2 ...

I think unnest is the only expr so far that represent a set of fields instead of singular value here, so i'm thinking of adding an extra check at this function https://github.com/apache/datafusion/blob/d58bae487329b7a7078429f083bffc611f42c8c7/datafusion/expr/src/utils.rs#L734